### PR TITLE
Make it possible to add builder-specific data to BuildResult.

### PR DESCRIPTION
**Pull request by iinozemtsev**
- Add builderDetails field to BuildResult
- Allow overriding serializers in client and server

### CODE REVIEW

**Comment by iinozemtsev** [comment]
I signed it!

**Comment by grouma** [comment]
Update the changelog and daemon version. I'll merge this and pull it into Google3.

File: build_daemon/lib/data/build_status.dart [hunk 1]
```diff
@@ -35,6 +35,10 @@ abstract class BuildResult implements Built<BuildResult, BuildResultBuilder> {
   String get error;
   @nullable
   bool get isCached;
+
+  /// Arbitrary additional information from daemon builder.
+  @nullable
+  Object get builderDetails;
```
Comment (grouma) [inline build_daemon/lib/data/build_status.dart]: If we are injecting custom serializers is this even necessary? Is it possible for the serializer to override this object and just work?

Comment (iinozemtsev) [reply build_daemon/lib/data/build_status.dart]: If I understand you correctly, this is basically composition vs inheritance (custom builders return BuildResult subclasses), and I don't have strong opinions regarding this, but usage of Built values add some nuances - BuildResult should be converted to an interface, and this affects builders generated for other built values which include BuildResult field.

I've created an alternative pull request #1959, it's hard to me to choose which approach is better without having other builder implementations in mind

**Comment by grouma** [comment]
The subclassing approach appears to be the better solution. Closing this one.

**Pull request closed without merging**